index.js

```diff
@@ -77,6 +77,45 @@ function validateCourse(course) {
     
 }
 
+
+/*  
+PUT method , URL:"/api/courses/id", filePath = `${__dirname}/courses.json`
+read data ==> modify it ==> write it back to the file system
+*/
+app.put('/api/courses/:id', (req,res) => {
+       
```

Comment:
@Eltayeb-Elgaali put method works fine but I am just curious and wondering the route you have wrote
 Normally we do not have path finish with courses.. we have a path ending with courses.json
let say if we do have courses.js , courses.md does it still work?Yeah I have tired it so it works. If you know how this route works maybe you can clarify it to me . Thx in advance :smile: